sound/soc/sof/ipc4-pcm.c

```diff
@@ -82,6 +132,14 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 	if (!pipeline_list->pipe_widgets)
 		return 0;
 
+	/* If this is a chained dma stream, then there is no need to continue further. */
```

Comment:
to be more explicit, we should say something like "Streams using chained DMA contain a single pipeline, so trigger that pipeline and return"